src/data/int/parity.lean

```diff
@@ -220,6 +220,21 @@ begin
     norm_num },
 end
 
+lemma even.two_mul_div_two : even n → 2 * (n / 2) = n :=
+by { rintro ⟨c, rfl⟩, simp }
+
+lemma even.div_two_mul_two : even n → n / 2 * 2 = n :=
+by { rintro ⟨c, rfl⟩, simp [mul_comm] }
+
+lemma odd.two_mul_div_two_add_one : odd n → 2 * (n / 2) + 1 = n :=
+by { rintro ⟨c, rfl⟩, rw mul_comm, convert int.div_add_mod' _ _, simp [int.add_mod], refl }
+
+lemma odd.div_two_mul_two_add_one : odd n → n / 2 * 2 + 1 = n :=
+by { rintro ⟨c, rfl⟩, convert int.div_add_mod' _ _, simp [int.add_mod], refl }
+
+lemma odd.add_one_div_two_mul_two : odd n → 1 + n / 2 * 2 = n :=
```

Comment:
This lemma seems a little redundant since you can prove it with `rw [add_comm, odd.div_two_mul_two_add_one]`.

I don't mind it being here, but I just wanted to know what the reasoning for having both was.

(I've asked on Zulip about having a `... = n - 1` variant, since that seems useful as a rw/simp lemma.)

Reply:
Agreed here

Reply:
> I don't mind it being here, but I just wanted to know what the reasoning for having both was.

We already have a similar situation in mathlib:

```lean
lemma nat.mod_add_div' (m k : ℕ) : m % k + (m / k) * k = m
lemma nat.div_add_mod' (m k : ℕ) : (m / k) * k + m % k = m
```

Reply:
but not with `1 +`